scc/src/opts.rs

```diff
@@ -49,86 +49,177 @@ pub fn fix_args(args: Vec<String>) -> Vec<String> {
 
 #[derive(Clone, Debug)]
 pub struct Opts {
-    pub script_paths: Vec<PathBuf>,
-    pub script_paths_file: Option<PathBuf>,
+    pub scripts_dir: Option<PathBuf>,
     pub cache_dir: Option<PathBuf>,
     pub optimize: bool,
     pub threads: u8,
-    pub no_warnings: bool,
+    pub warnings: Vec<String>,
     pub no_testonly: bool,
     pub no_breakpoint: bool,
-    pub profile_off: bool,
+    pub profile: bool,
+    pub script_paths_file: Option<PathBuf>,
     pub cache_file: Option<PathBuf>,
 }
 
-fn toggle_options(name: &'static str, help: &'static str) -> impl Parser<bool> {
+impl Opts {
+    pub const fn default() -> Self {
+        Self {
+            scripts_dir: None,
+            cache_dir: None,
+            optimize: false,
+            threads: 1,
+            warnings: vec![],
+            no_testonly: false,
+            no_breakpoint: false,
+            profile: true,
+            script_paths_file: None,
+            cache_file: None,
+        }
+    }
+}
+
+trait SlongArg {
+    fn is_not_slong(&self) -> bool;
+}
+
+impl SlongArg for PathBuf {
+    fn is_not_slong(&self) -> bool {
+        matches!((*self).to_str(), Some(str) if !str.starts_with('-'))
+    }
+}
```

Comment:
I don't see why this is a trait, it's simpler to just make it a function and pass it directly
```rs
.guard(is_not_slong, "starts with -")
```